File: code-coaches/hard/hofstadters-q-sequence/HofstadtersQSequence.cs
using System;

class HofstadtersQSequence
{
    static int QSequence(int n)
    {
        if (n <= 2)
            return 1;
        else
            return QSequence(n - QSequence(n - 1)) + QSequence(n - QSequence(n - 2));
    }

    static void Main()
    {
        int n = int.Parse(Console.ReadLine());
        Console.WriteLine(QSequence(n));
    }
}

File: code-coaches/hard/hofstadters-q-sequence/HofstadtersQSequence.swift
func qSequence(_ n: Int) -> Int {
    if n <= 2 {
        return 1
    } else {
        return qSequence(n - qSequence(n - 1)) + qSequence(n - qSequence(n - 2))
    }
}

func main() {
    let n = Int(readLine()!)!
	print(qSequence(n))
}

main()